src/metrics/handle.rs

```diff
@@ -2,6 +2,43 @@ use metrics::GaugeValue;
 use std::slice;
 use std::sync::atomic::{AtomicU32, AtomicU64, Ordering};
 
+#[derive(Debug)]
+struct AtomicF64 {
+    inner: AtomicU64,
+}
+
+impl AtomicF64 {
+    fn new(init: f64) -> Self {
+        Self {
+            inner: AtomicU64::new(init.to_bits()),
+        }
+    }
+
+    pub fn fetch_update<F>(
+        &self,
+        set_order: Ordering,
+        fetch_order: Ordering,
+        mut f: F,
+    ) -> Result<f64, f64>
+    where
+        F: FnMut(f64) -> Option<f64>,
+    {
+        let res = self.inner.fetch_update(set_order, fetch_order, |x| {
+            let opt: Option<f64> = f(f64::from_bits(x));
+            opt.map(|i| i.to_bits())
+        });
+
+        match res {
+            Ok(f) => Ok(f64::from_bits(f)),
+            Err(f) => Ok(f64::from_bits(f)),
+        }
```

Comment:
perhaps:
```rust
        Ok(f64::from_bits(res.unwrap_or_else(|f| f)))
```

Reply:
Oops, this is actually a bug. This should return `Err` in the second leg of the match.

Reply:
Well then?
```rust
        res.map(f64::from_bits).map_err(f64::from_bits)
```
(not that it really matters :grin:)

Reply:
That looks tidier to me!